OsmAnd/src/net/osmand/plus/mapcontextmenu/builders/AmenityMenuBuilder.java

```diff
@@ -674,6 +682,60 @@ private String getFormattedInt(String value) {
 		}
 	}
 
+	private String[] getFormattedPrefixAndText(String key, String prefix, String value) {
+		DecimalFormat df = new DecimalFormat("#.##");
+		df.setRoundingMode(RoundingMode.CEILING);
+		switch (key){
+			case "width":
+			case "height":
+				prefix = Algorithms.capitalizeFirstLetter(key);
+			case "depth":
+			case "seamark_height":
+				if (metricSystem == OsmandSettings.MetricsConstants.MILES_AND_FEET) {
+					value = String.valueOf(df.format(Double.valueOf(value)*OsmAndFormatter.FEET_IN_ONE_METER))
+						+ " " + mapActivity.getResources().getString(R.string.foot);
+				} else if (metricSystem == OsmandSettings.MetricsConstants.MILES_AND_YARDS) {
+					value = String.valueOf(df.format(Double.valueOf(value)*OsmAndFormatter.YARDS_IN_ONE_METER))
+						+ " "+ mapActivity.getResources().getString(R.string.yard);
+				} else {
+					value = value + " " + mapActivity.getResources().getString(R.string.m);
+				}
+				break;
+			case "distance":
+				if (metricSystem == OsmandSettings.MetricsConstants.MILES_AND_FEET ||
+						metricSystem == OsmandSettings.MetricsConstants.MILES_AND_YARDS ||
+						metricSystem == OsmandSettings.MetricsConstants.MILES_AND_METERS) {
+					value = String.valueOf(df.format(Double.valueOf(value)*1000/OsmAndFormatter.METERS_IN_ONE_MILE))
```

Comment:
do Double.valueOf(value) * 1000 once only, use spaces between operands
btw - we have getFormattedDistance, use it

Reply:
Changed to getFormattedDistance for distance but retained local conversion for height/length/width.